src/System.CommandLine/README.md

```diff
@@ -0,0 +1,227 @@
+# System.CommandLine
+
+System.CommandLine provides robust support for command-line parsing, invocation, and shell completions in .NET applications. It supports both POSIX and Windows conventions, making it easy to build professional command-line interfaces.
+
+## Getting Started
+
+### Basic Command
+
+Here's a simple "Hello World" command-line application:
+
+```csharp
+using System.CommandLine;
+
+var rootCommand = new RootCommand("Sample command-line app");
+
+var nameOption = new Option<string>(
+    aliases: ["--name", "-n"],
+    description: "Your name");
+
+rootCommand.Options.Add(nameOption);
+
+rootCommand.SetAction(parseResult =>
+{
+    string name = parseResult.GetValueForOption(nameOption);
+    Console.WriteLine($"Hello, {name ?? "World"}!");
+});
+
+return await rootCommand.InvokeAsync(args);
+```
+
+### Commands with Arguments
+
+Arguments are values passed directly to commands without option names:
+
+```csharp
+var fileArgument = new Argument<FileInfo>(
+    name: "file",
+    description: "The file to process");
+
+var processCommand = new Command("process", "Process a file");
+processCommand.Arguments.Add(fileArgument);
+
+processCommand.SetAction(parseResult =>
+{
+    FileInfo file = parseResult.GetValueForArgument(fileArgument);
+    Console.WriteLine($"Processing {file.FullName}");
+});
+
+rootCommand.Subcommands.Add(processCommand);
+```
+
+### Options with Default Values
+
+Options can have default values and validation:
+
+```csharp
+var delayOption = new Option<int>(
+    aliases: ["--delay", "-d"],
+    getDefaultValue: () => 1000,
+    description: "Delay in milliseconds");
+
+delayOption.AddValidator(result =>
+{
+    if (result.GetValueOrDefault<int>() < 0)
+    {
+        result.ErrorMessage = "Delay must be non-negative";
+    }
+});
+
+rootCommand.Options.Add(delayOption);
+```
+
+### Subcommands
+
+Build complex CLI applications with nested commands:
+
+```csharp
+var rootCommand = new RootCommand("My application");
+
+var configCommand = new Command("config", "Configure the application");
+var configSetCommand = new Command("set", "Set a configuration value");
+var configGetCommand = new Command("get", "Get a configuration value");
+
+var keyOption = new Option<string>("--key", "Configuration key");
```

Comment:
The second parameter here will be applied to the `params string[] aliases` parameter.